### PR TITLE
[Merged by Bors] - Refactor(Algebra/Group/Subgroup/Finite): Replace `Fintype.card` with `Nat.card`

### PR DESCRIPTION
**Pull request by tb65536**
A lot of the group theory library still needs to be switched over from `Fintype.card` to `Nat.card`. This PR switches over `Algebra/Group/Subgroup/Finite.lean`. I had to add a little noise to a few other files, but these will get cleaned up once those files are switched over later.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean [hunk 1]
```diff
@@ -131,14 +131,20 @@ theorem isCyclic_of_orderOf_eq_card [Fintype α] (x : α) (hx : orderOf x = Fint
...
+    (H : Subgroup G) {p : ℕ} [hp : Fact p.Prime] (h : Fintype.card G = p) : H = ⊥ ∨ H = ⊤ := by
+  classical
+  have := card_subgroup_dvd_card H
+  rwa [h, Nat.dvd_prime hp.1, ← h, ← Nat.card_eq_fintype_card, ← Nat.card_eq_fintype_card,
+    ← eq_bot_iff_card, card_eq_iff_eq_top] at this
+
 /-- Any non-identity element of a finite group of prime order generates the group. -/
 @[to_additive "Any non-identity element of a finite group of prime order generates the group."]
 theorem zpowers_eq_top_of_prime_card {G : Type*} [Group G] {_ : Fintype G} {p : ℕ}
     [hp : Fact p.Prime] (h : Fintype.card G = p) {g : G} (hg : g ≠ 1) : zpowers g = ⊤ := by
```
Comment (alreadydone) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:145]: What do you think about assuming `Fact (Fintype.card G).Prime` and get rid of `p`?
```suggestion
theorem Subgroup.eq_bot_or_eq_top_of_prime_card {G : Type*} [Group G] {_ : Fintype G}
    (H : Subgroup G) {p : ℕ} [hp : Fact p.Prime] (h : Fintype.card G = p) : H = ⊥ ∨ H = ⊤ := by
  classical
  have := card_subgroup_dvd_card H
  rwa [h, Nat.dvd_prime hp.1, ← h, ← Nat.card_eq_fintype_card, ← Nat.card_eq_fintype_card,
    ← eq_bot_iff_card, card_eq_iff_eq_top] at this

/-- Any non-identity element of a finite group of prime order generates the group. -/
@[to_additive "Any non-identity element of a finite group of prime order generates the group."]
theorem zpowers_eq_top_of_prime_card {G : Type*} [Group G] {_ : Fintype G} {p : ℕ}
    [hp : Fact p.Prime] (h : Fintype.card G = p) {g : G} (hg : g ≠ 1) : zpowers g = ⊤ := by
```

**Review by alreadydone — APPROVED** [review]
Thanks, LGTM

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean (hunk 1 above)
Comment (tb65536) [reply Mathlib/GroupTheory/SpecificGroups/Cyclic.lean]: Good point. I've made this change in the new helper lemma `eq_bot_or_eq_top_of_prime_card`, but making this change throughout the whole file should probably wait for another PR (maybe the one where I switch this file over to `Nat.card`).

File: Mathlib/GroupTheory/SpecificGroups/Cyclic.lean [hunk 2]
```diff
@@ -517,12 +518,12 @@ alias IsAddCyclic.card_orderOf_eq_totient := IsAddCyclic.card_addOrderOf_eq_toti
...
 @[to_additive "A finite group of prime order is simple."]
 theorem isSimpleGroup_of_prime_card {α : Type u} [Group α] [Fintype α] {p : ℕ} [hp : Fact p.Prime]
-    (h : Fintype.card α = p) : IsSimpleGroup α :=
+    (h : Fintype.card α = p) : IsSimpleGroup α := by
+  subst h
   have : Nontrivial α := by
-    have h' := Nat.Prime.one_lt (Fact.out (p := p.Prime))
-    rw [← h] at h'
+    have h' := Nat.Prime.one_lt hp.out
     exact Fintype.one_lt_card_iff_nontrivial.1 h'
```
Comment (alreadydone) [inline Mathlib/GroupTheory/SpecificGroups/Cyclic.lean:525]: I guess this works but let's leave it to a future PR.
```suggestion
  have : Nontrivial α := Fintype.one_lt_card_iff_nontrivial.1 hp.out.one_lt
```

**Review by alreadydone — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**